packages/image/README.md

```diff
@@ -0,0 +1,108 @@
+};
+
+export default config;
+```
+
+## Usage
+
+### When using one of the providers, i.e. an image CDN:
+
+```svelte
+<script>
+	import Image from '@sveltejs/image';
+</script>
+
+<Image src="/path/to/your/image.jpg" width={1200} height={1800} alt="An alt text" />
+```
+
+`width` and `height` should be the natural width/height of the referenced image. `alt` should describe the image. All are required. The `src` is transformed by calling `getURL` of the `default` provider provided in the `vite.config.js`.
+
+### Static build time optimization:
+
+```svelte
+<script>
+	import Image from '@sveltejs/image';
+	import YourImage from '/path/to/your/image.jpg';
+</script>
+
+<Image src={YourImage} alt="An alt text" />
+```
+
+This optimizes the image at build time using `vite-imagetools`. `width` and `height` are optional as they can be infered from the source image.
+
+### Pros/Cons of the solutions
+
+Using the static provider generates the images at build time, so that build time may take longer the more images you transform.
+
+Using an image CDN provides more flexibility with regards to sizes and you can pass image sources not known at build time, but it comes with potentially a bit setup overhead (configuring the image CDN) and possibly usage cost.
+
+You can mix and match both solutions in one project.
+
+### `Image` component options
+
+There are a few things you can customize:
+
+- `priority`: give this to the most important/largest image on the page so it loads faster
+- `sizes`: If your image is less than full width on one or more screen sizes, add this info here. When using dynamic providers the widths can be adjusted accordingly to produce a more optimal `srcset`. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/sizes) for more info on the attribute
+- `style`: to style the image
+- `class`: to style the image. Be aware that you need to pass classes that are global (i.e. wrapped in `:global()` when coming from a `<style>` tag)
+- `loading`: loading behavior of the image. Defaults to `lazy` which means it's loaded only when about to enter the viewport. Set to `eager` to load right away (the default when setting `priority`)
+- `provider`: you can pass more than the `default` provider in `vite.config.js`. If you then want to use a different provider than the `default` one, pass it here
+- `providerOptions`: provider-specific image CDN options. For example `quality` for Vercel
+
+## Best practices
+
+- Always provide a good `alt` text
+- Your original images should have a good quality/resolution. Images are easier to size down than up
+- Choose one image per page which is the most important/largest one and give it `priority` so it loads faster. This gives you better web vitals scores (largest contentful paint in particular)
+- Give the image a container or a styling so that it is constrained and does not jump around. `width` and `height` help the browser reserving space while the image is still loading
+
+## Roadmap
```

Comment:
I didn't add the backticks around the code as it'd end the suggestion and I wasn't quite sure how to escape
```suggestion
## Automatic imports

You can use [svelte-preprocess-import-assets](https://github.com/bluwy/svelte-preprocess-import-assets) to automatically import your images:

import { vitePreprocess } from '@sveltejs/kit/vite';
import importAssets from 'svelte-preprocess-import-assets';

/** @type {import('@sveltejs/kit').Config} */
export default {
	preprocess: [
		importAssets({
			sources: (defaultSources) => {
				return [
					...defaultSources,
					{
						tag: 'Image',
						srcAttributes: ['src']
					}
				]
			}
		}),
		vitePreprocess()
	]
...


Then, rather than importing your image, you can simply write:
<Image src="./myimage.png" />
```